artifactory.py

```diff
@@ -1779,15 +1755,17 @@ def deploy_by_checksum(
         artifact content already exists in Artifactory.
 
         :param pathobj: ArtifactoryPath object
-        :param checksum_sha1: sha1Value
-        :param checksum_sha256: sha256Value
-        :param checksum: checksum value (type is resolved by length)
+        :param sha1: sha1Value
+        :param sha256: sha256Value
+        :param checksum: sha1Value or sha256Value
```

Comment:
even if it is a wrapper, do we really need 1:1 match?
for example `deploy_file()` calculates checksum for the user

probably we can follow here the same:
```suggestion
        :param sha1: (bool) calculate sha1 for the object, default True
        :param sha256: (bool) calculate sha256 for the object, default True
```

if both are false, then raise an error
and `checksum` could be removed, since here we identify checksum header

Reply:
@beliaev-maksim 
Thanks for your advice.

If we use `deploy_by_checksum`, we know the sha value already. 

Consider the following scenario,
I have a long sha list, including sha1 and sha256, if `checksum` exists, the snippet would be
```
for sha in sha_list:
    deploy_by_checksum(checksum=sha)
``` 
In the contrast,
```
for sha in sha_list:
    if len(sha) == 40:
        deploy_by_checksum(sha1=sha)
    else:
        deploy_by_checksum(sha256=sha)
```

The first one is easier to use, so we need `checksum`. Do you have concern?

Reply:
yes, really relevant scenario 

and if needed, then user will calculate checksum manually with `artifactory.sha1sum()`

can you add these notes to the help as well?

Reply:
If the user know sha value already, why do they need to calculate checksum again when get help from `deploy_by_checksum`?

Reply:
@zhan9san 
I mean add example in README.md showing the test case

Reply:
No problem. Will add some examples in README.

Reply:
Hi @beliaev-maksim 

Examples are added in README. All commits are squashed. Could you please review it again?